src/close.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{transaction::CloseTransaction, Channel};
+use bitcoin::{Address, Transaction};
+use ecdsa_fun::Signature;
+
+pub struct State0(Channel);
+pub struct State1 {
+    channel: Channel,
+    final_address_self: Address,
+    final_address_other: Address,
+}
+pub struct FinalState(Transaction);
+
+impl FinalState {
+    pub fn into_transaction(self) -> Transaction {
+        self.0
+    }
+}
+
+impl From<Channel> for State0 {
+    fn from(channel: Channel) -> Self {
+        Self(channel)
+    }
+}
+
+pub struct Message0 {
+    final_address: Address,
+}
+
+#[derive(Debug)]
+pub struct Message1 {
+    close_transaction: CloseTransaction,
+    sig_close_transaction: Signature,
+}
+
+impl State0 {
+    pub fn compose(&self, final_address: Address) -> Message0 {
```

Comment:
:+1: I like the ability to spend directly to a wallet address. It's also nice because at a higher level (the eventual close function on `Channel`) we can decide to either use a pre-generated address or ask the wallet for a new address before starting the close protocol.